hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/table/action/commit/UpsertPartitioner.java

```diff
@@ -235,6 +243,7 @@ private void assignInserts(WorkloadProfile profile, HoodieEngineContext context)
         LOG.info("Total insert buckets for partition path " + partitionPath + " => " + insertBuckets);
         partitionPathToInsertBucketInfos.put(partitionPath, insertBuckets);
       }
+      profile.getOutputPartitionPathStatMap().putIfAbsent(partitionPath, outputWorkloadStats);
```

Comment:
Why `putIfAbsent`? Should we not update the stat in this case?

Reply:
good catch. will fix it.